build/lib/src/builder/build_step_impl.dart

```diff
@@ -59,26 +43,25 @@ class BuildStepImpl implements BuildStep {
   /// The current root package, used for input/output validation.
   final String _rootPackage;
 
-  BuildStepImpl(this.input, Iterable<AssetId> expectedOutputs, this._reader,
-      this._writer, this._rootPackage, this._resolvers)
-      : expectedOutputs = new List.unmodifiable(expectedOutputs) {
-    /// The [input] is always a dependency.
-    _dependencies.add(input.id);
+  BuildStepImpl(Asset input, Iterable<AssetId> expectedOutputs, this._reader,
+      this._writer, this._rootPackage, this._resolvers,
+      {Logger logger})
+      : this.input = input,
+        this.logger = logger ?? new Logger('${input.id}') {
+    _expectedOutputs.addAll(expectedOutputs);
```

Comment:
It's not exposed outside this library anymore and I generally like UnmodifiableList as a signal that we're being defensive because it's visible to other code